src/modm/driver/temperature/mcp990x.hpp

```diff
@@ -0,0 +1,131 @@
+/*
+* Copyright (c) 2022, Christopher Durand
+*
+* This file is part of the modm project.
+*
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_MCP990X_HPP
+#define MODM_MCP990X_HPP
+
+#include <modm/architecture/interface/register.hpp>
+#include <modm/architecture/interface/i2c_device.hpp>
+#include <array>
+
+namespace modm
+{
+
+/// @ingroup modm_driver_mcp990x
+struct mcp990x
+{
+protected:
+	/// @cond
+	enum class
+	Register : uint8_t
+	{
+		IntTempHigh	= 0x00,
+		Config		= 0x03,
+		IntTempLow	= 0x29,
+		ProductId	= 0xfd
+	};
+
+	enum class
+	Config : uint8_t
+	{
+		MaskAlert = Bit7,
+		Standby = Bit6,
+		AlertMode = Bit5,
+		DisableErrorCorrectionDiode1 = Bit4,
+		DisableErrorCorrectionDiode2 = Bit3,
+		IfAddInc = Bit3,
+		ExtendedRange = Bit2,
+		DisableDynamicAveraging = Bit1,
+		DisableAntiParallelDiodeMode = Bit0
+	};
+	MODM_FLAGS8(Config);
+	/// @endcond
+public:
+	struct modm_packed
+	Data
+	{
+		static constexpr auto ExtendedOffset{64}; // °C
+		uint8_t data[2];
+
+		/// @return temperature in 0.125 °C steps
+		constexpr int16_t
+		getTemperatureFractional()
+		{
+			return ((data[1] << 3) | (data[0] >> 5)) - (ExtendedOffset * 8);
+		}
```

Comment:
Only true in _Extended Range Mode_, but since this mode gets configured by this driver and the configuration is not user accessible: Fine for me.

Reply:
There is no real reason to not enable extended range mode when the sensor is used standalone with modm. This part normally goes onto PC main boards and the other mode is probably there for compatibility with chipsets. In both modes data is 11 bits and 0.125 °C resolution. You would only save one subtraction.